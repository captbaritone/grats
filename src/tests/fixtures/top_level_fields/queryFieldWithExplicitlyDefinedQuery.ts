/** @gqlQueryField */
export function greeting(): string {
  return "Hello world";
}

/**
 * I might want to explicitly define a type here to provide a description.
 *
 * @gqlType */
export type Query = unknown;
