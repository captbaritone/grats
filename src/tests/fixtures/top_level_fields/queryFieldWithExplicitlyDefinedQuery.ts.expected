-----------------
INPUT
----------------- 
/** @gqlQueryField */
export function greeting(): string {
  return "Hello world";
}

/**
 * I might want to explicitly define a type here to provide a description.
 *
 * @gqlType */
export type Query = unknown;

-----------------
OUTPUT
-----------------
-- SDL --
"""
I might want to explicitly define a type here to provide a description.
"""
type Query {
  greeting: String
}
-- TypeScript --
import { greeting as queryGreetingResolver } from "./queryFieldWithExplicitlyDefinedQuery";
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const QueryType: GraphQLObjectType = new GraphQLObjectType({
        name: "Query",
        description: "I might want to explicitly define a type here to provide a description.",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString,
                    resolve() {
                        return queryGreetingResolver();
                    }
                }
            };
        }
    });
    return new GraphQLSchema({
        query: QueryType,
        types: [QueryType]
    });
}
